Sources/MongoProvider/Provider.swift

```diff
@@ -36,20 +36,60 @@ public final class Provider: Vapor.Provider {
 
 extension MongoDriver: ConfigInitializable {
     public convenience init(config: Config) throws {
+        
         guard let mongo = config["mongo"] else {
             throw ConfigError.missingFile("mongo")
         }
         
-        guard let url = mongo["url"]?.string else {
-            throw ConfigError.missing(
-                key: ["url"],
-                file: "mongo",
-                desiredType: String.self
-            )
+        func buildUrl(mongo: Config) throws -> String {
```

Comment:
The config file could be mapped to [ClientSettings](https://github.com/OpenKitten/MongoKitten/blob/mongokitten4/Sources/MongoKitten/Server/ClientSettings.swift). That simplifies this function and improves compatibility with edge-case scenarios such as complex passwords with `%` and other URL reserved characters.